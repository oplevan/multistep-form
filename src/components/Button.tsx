import * as React from "react";
import { cn } from "../utils/cn";

interface ButtonProps extends React.ButtonHTMLAttributes<HTMLButtonElement> {}

export default function Button({ children, className, type, ...props }: ButtonProps) {
  return (
    <button
      type={type}
      className={cn(
        "px-4 py-1 text-sm uppercase font-semibold rounded-md ring text-white bg-blue-600 shadow-lg hover:bg-blue-700",
        className
      )}
      {...props}
    >
      {children}
    </button>
  );
}
