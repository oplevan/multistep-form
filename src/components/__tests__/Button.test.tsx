import { render, screen } from "@testing-library/react";
import Button from "../Button";

describe("<Button /> component", () => {
  test("should render correctly", () => {
    const { container } = render(<Button />);
    expect(container).toMatchSnapshot();
  });

  test("should have a custom class name 'test'", () => {
    render(<Button className="test" />);
    const button = screen.getByRole("button");
    expect(button).toBeInTheDocument();
    expect(button).toHaveClass("test");
  });

  test("should have correct type attribute", () => {
    render(<Button type="submit" />);
    const button = screen.getByRole("button");
    expect(button).toBeInTheDocument();
    expect(button).toHaveAttribute("type", "submit");
  });

  test("should have a custom attribute", () => {
    render(<Button custom-attribute="test" />);
    const button = screen.getByRole("button");
    expect(button).toBeInTheDocument();
    expect(button).toHaveAttribute("custom-attribute", "test");
  });

  test("should have the correct name", () => {
    render(<Button>TEST BUTTON</Button>);
    const button = screen.getByRole("button");
    expect(button).toBeInTheDocument();
    expect(button).toHaveTextContent("TEST BUTTON");
  });

  test("should render children inside a div", () => {
    render(
      <Button>
        <div className="children-wrapped-in-a-div">TEST BUTTON</div>
      </Button>
    );
    const button = screen.getByRole("button");
    expect(button).toBeInTheDocument();
    expect(button).toContainHTML("<div class='children-wrapped-in-a-div'>TEST BUTTON</div>");
  });
});
