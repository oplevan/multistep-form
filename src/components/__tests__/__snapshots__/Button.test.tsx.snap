// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> component matches snapshot 1`] = `
<div>
  <button
    class="px-4 py-1 text-sm uppercase font-semibold rounded-md ring text-white bg-blue-600 shadow-lg hover:bg-blue-700"
  />
</div>
`;

exports[`<Button /> component should render correctly 1`] = `
<div>
  <button
    class="px-4 py-1 text-sm uppercase font-semibold rounded-md ring text-white bg-blue-600 shadow-lg hover:bg-blue-700"
  />
</div>
`;
